cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexUpdaterTest.java

```diff
@@ -125,6 +127,123 @@ public void updateOneFieldIndex() throws Exception {
+                    rev.docId = String.format("id_%d_%s",i,Thread.currentThread().getName());
+                    System.out.println("creating "+rev.docId);
+                    Map<String, Object> bodyMap = new HashMap<String, Object>();
+                    if (i % 2 == 0) {
+                        bodyMap.put("name", "mike");
+                    } else {
+                        bodyMap.put("name", "tom");
+                    }
+                    rev.body = DocumentBodyFactory.create(bodyMap);
+                    try {
+                        ds.createDocumentFromRevision(rev);
+                    } catch (DocumentException de) {
+                        Assert.fail("Exception thrown when creating revision " + de);
+                    }
+                    // batch up index updates
+                    if (i % nUpdates == 0) {
+                        IndexUpdater.updateIndex("basic", fields, db, ds, im.getQueue());
+                    }
+                }
+            }
+        }
+
+        List<Thread> threads = new ArrayList<Thread>();
+
+        // Create, start and wait for the threads to complete
+        for (int i = 0; i < nThreads; i++) {
+            threads.add(new PopulateThread());
+        }
+        for (Thread t : threads) {
+            t.start();
+            // try to get all threads to start simultaneously
+            latch.countDown();
+        }
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        // catch any missing index updates
+        IndexUpdater.updateIndex("basic", fields, db, ds, im.getQueue());
+
+        // now the "basic" index should be up to date, check the sequence number
+        // and the values in the index
+
+        assertThat(getIndexSequenceNumber("basic"), is((long)nDocs*nThreads));
+
+        int mikes = 0;
+        int toms = 0;
+
+        cursor = null;
+        try {
+            SQLDatabase db = TestUtils.getDatabaseConnectionToExistingDb(this.db);
+            cursor = db.rawQuery(sql, new String[]{});
+            assertThat(cursor.getCount(), is(nDocs*nThreads));
+            assertThat(cursor.getColumnCount(), is(3));
+            while (cursor.moveToNext()) {
+                assertThat(cursor.columnName(0), is("_id"));
+                assertThat(cursor.columnName(1), is("_rev"));
+                assertThat(cursor.columnName(2), is("name"));
+                int docNum = Integer.valueOf(cursor.getString(0).split("_")[1]);
+                // even document numbers should have name==mike, odd name==tom
+                if (docNum % 2 == 0) {
+                    assertThat(cursor.getString(2), is("mike"));
+                    mikes++;
+                } else {
+                    assertThat(cursor.getString(2), is("tom"));
+                    toms++;
+                }
+            }
+            // check that the number of values on the name field is correct
+            assertThat(mikes, is(nDocs*nThreads/2));
```

Comment:
Whilst these assertions are ok now, this first one could easily break if someone changes `nDocs` or `nThreads` to produce an odd multiple. I think 
`(nDocs*nThreads/2) + (nDoc*nThreads % 2)` always works.